packages/web/src/components/SignUpForm/SignUpForm.tsx

```diff
@@ -16,29 +16,101 @@ type Props = {
 
 export const SignUpForm = ({ isLoading, error, onSubmit }: Props) => {
   const [firstName, setFirstName] = useState('');
+  const [firstNameError, setFirstNameError] = useState('');
+  const [firstNameTouched, setFirstNameTouched] = useState(false);
+
   const [lastName, setLastName] = useState('');
+  const [lastNameError, setLastNameError] = useState('');
+  const [lastNameTouched, setLastNameTouched] = useState(false);
+
   const [email, setEmail] = useState('');
+  const [emailError, setEmailError] = useState('');
+  const [emailTouched, setEmailTouched] = useState(false);
+
   const [password, setPassword] = useState('');
+  const [passwordError, setPasswordError] = useState('');
+  const [passwordTouched, setPasswordTouched] = useState(false);
+
   const [passwordMatch, setPasswordMatch] = useState('');
+  const [passwordMatchError, setPasswordMatchError] = useState('');
+  const [passwordMatchTouched, setPasswordMatchTouched] = useState(false);
+
   const [isValid, setIsValid] = useState(false);
-  const [isPasswordMatch, setIsPasswordMatch] = useState(true);
+
+  // I've decided to keep it here for now, but later probably we can use 'validate' or/and 'Formik' npm libs.
+  const validateEmail = (emailToTest) => {
+    // eslint-disable-next-line
+    const re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
```

Comment:
How about move this to a separate regexp.constant file?

Reply:
See the comment above the function.

Reply:
This asks at least for a separate hook and a combined object, but let's keep it quick and dirty for now if it works.